graylog2-server/src/main/java/org/graylog2/buffers/processors/ServerProcessBufferProcessor.java

```diff
@@ -41,10 +42,10 @@
 
     @Inject
     public ServerProcessBufferProcessor(MetricRegistry metricRegistry,
-                                        OrderedMessageProcessors orderedMessageProcessors,
+                                        Provider<OrderedMessageProcessors> orderedMessageProcessorsProvider,
                                         OutputBuffer outputBuffer) {
         super(metricRegistry);
-        this.orderedMessageProcessors = orderedMessageProcessors;
+        this.orderedMessageProcessors = orderedMessageProcessorsProvider.get();
```

Comment:
I was being extra-super-duper careful and explicit, but you are correct, it is not strictly necessary.